packages/docs-website/docs/protocol-docs/client-specification/json-rpc-api.md

```diff
@@ -0,0 +1,420 @@
+
+| Name       | Type            | Description                                                            |
+| ---------- | --------------- | ---------------------------------------------------------------------- |
+| domain     | string          | The channel the funds are attributed to                                |
+| hubAddress | string          | Signing address of channel intermediary                                |
+| budgets    | `TokenBudget[]` | An array of approved send and received capacity for a list of channels |
+
+### Errors
+
+TODO: (HIGH) Define errors in typescript
+
+## UpdateChannel
+
+Used to take a turn in your channel, or propose an update to a channel or application state. The application update must be a valid transition, and properly signed. The updated `ChannelResult` is returned.
+
+### Parameters
+
+| Name        | Type           | Description                        |
+| ----------- | -------------- | ---------------------------------- |
+| channelId   | `Byttes32`     | Derived channel identifier         |
+| allocations | `Allocation[]` | Updated token balances for channel |
+| appData     | string         | Encoded updated application state  |
+
+### Response
+
+| Name                    | Type              | Description                                |
+| ----------------------- | ----------------- | ------------------------------------------ |
+| participants            | `Participant[]`   | Identifying information members of channel |
+| allocations             | `Allocation[]`    | Array of funding amounts for participants  |
+| appData                 | string            | Encoded initial state of app               |
+| appDefinition           | string            | Address of contract governing the channel  |
+| channelId               | string            | Unique channel identifier                  |
+| status                  | `ChannelStatus`   | Current status of channel                  |
+| turnNum                 | number            | Channel nonce                              |
+| challengeExpirationTime | number (optional) | Time current challenge on channel elapses  |
+
+### Errors
+
+TODO: (HIGH) Clean up error message types and codes
+
+| Code | Message            | Description                                 |
+| ---- | ------------------ | ------------------------------------------- |
+| 400  | Channel not found  | Could not find channel to update in storage |
+| 401  | Invalid transition | Illegal state transition proposed           |
+| 402  | Invalid app data   | Incorrect encoded app information           |
+| 403  | Not your turn      | Cannot update channel                       |
+| 404  | Channel closed     | Channel no longer accepting updates         |
+
+## JoinChannel
+
+Called when you would are joining a channel that has been created. Generally, creating a channel is done, and a notification is sent to the desired counterparty, who then dispatches the `JoinChannel` request.
```

Comment:
When `createChannel` is called, that wallet will queue messages for the other participants. Once these are delivered to the other wallets, they will emit a `ChannelProposed` notification. The application itself should decide what to do (most likely, call `joinChannel`). 

Hopefully it is clear [here](https://docs.statechannels.org/docs/app-devs/make-api-calls).